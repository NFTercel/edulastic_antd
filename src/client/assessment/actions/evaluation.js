import { CHECK_ANSWER } from "../constants/actions";

export const evaluateAnswer = () => ({
  type: CHECK_ANSWER
});
