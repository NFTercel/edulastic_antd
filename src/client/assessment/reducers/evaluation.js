import { ADD_ITEM_EVALUATION } from "../constants/actions";

const initialState = {};

const evaluation = (state = initialState, { type, payload }) => {
  switch (type) {
    case ADD_ITEM_EVALUATION:
      return {
        ...payload
      };
    default:
      return state;
  }
};

export default evaluation;
