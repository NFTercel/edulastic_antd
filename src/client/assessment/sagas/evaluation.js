import { takeEvery, put, all, select, call } from "redux-saga/effects";
import { message } from "antd";
import { testItemsApi } from "@edulastic/api";

// actions
import { CHECK_ANSWER_EVALUATION, ADD_ITEM_EVALUATION, CHANGE_PREVIEW } from "../constants/actions";
import { itemQuestionsSelector, answersForCheck } from "../selectors/test";

function* evaluateAnswers() {
  try {
    const questionIds = yield select(itemQuestionsSelector);
    const allAnswers = yield select(answersForCheck);
    const answerIds = Object.keys(allAnswers);
    const userResponse = {};

    answerIds.forEach(id => {
      if (questionIds.includes(id)) {
        userResponse[id] = allAnswers[id];
      }
    });

    const { items, currentItem } = yield select(state => state.test);
    const id = items[currentItem]._id;
    const result = yield call(testItemsApi.evaluation, id, userResponse);

    yield put({
      type: CHANGE_PREVIEW,
      payload: {
        view: "check"
      }
    });

    yield put({
      type: ADD_ITEM_EVALUATION,
      payload: {
        ...result
      }
    });

    yield put({
      type: ADD_ITEM_EVALUATION,
      payload: {
        ...result.evaluations
      }
    });
    const msg = `score: ${result.score} / ${result.maxScore}`;
    yield call(message.success, msg, 0.5);
  } catch (err) {
    console.log(err);
  }
}

export default function* watcherSaga() {
  yield all([yield takeEvery(CHECK_ANSWER_EVALUATION, evaluateAnswers)]);
}
