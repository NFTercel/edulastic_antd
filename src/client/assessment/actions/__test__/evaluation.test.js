import { evaluateAnswer } from "../evaluation";

describe("Assessment", () => {
  it("assessment should return an action", () => {
    expect(evaluateAnswer({})).toMatchSnapshot();
  });
});
