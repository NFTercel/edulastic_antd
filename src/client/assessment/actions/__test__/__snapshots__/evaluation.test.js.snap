// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment assessment should return an action 1`] = `
Object {
  "type": "[answers] check answers",
}
`;
